### PR TITLE
Use barriers to reduce duplicate work/resources

### PR DESCRIPTION
**Pull request by jysohn23**

### CODE REVIEW

File: transformers/modeling_utils.py [hunk 1]
```diff
@@ -245,17 +245,20 @@ def save_pretrained(self, save_directory, xla_device=False):
...
+
+        # Save configuration file
         if xla_device:
-            # Saving for each process is save since output_dir is proc# namespaced.
             import torch_xla.core.xla_model as xm
-            xm.save(model_to_save.state_dict(), output_model_file, master_only=False)
+            if xm.is_master_ordinal():
+                model_to_save.config.save_pretrained(save_directory)
+            # xm.save takes care of saving only from master
+            xm.save(model_to_save.state_dict(), output_model_file)
```
Comment (taylanbil) [inline transformers/modeling_utils.py:257]: idea for the future; you could use the checkpoint tagger later, if they mark which chpt is the best etc.

File: examples/run_glue_tpu.py [hunk 2]
```diff
@@ -341,6 +355,9 @@ def main(args):
     label_list = processor.get_labels()
     num_labels = len(label_list)
 
+    if not xm.is_master_ordinal():
+        xm.rendezvous('download_only_once')  # Make sure only the first process in distributed training will download model & vocab
```
Comment (taylanbil) [inline examples/run_glue_tpu.py:359]: I'm assuming no tensor work is done in the methods b/w `download_only_once` rendezvous's

**Review by taylanbil — APPROVED** [review]
much cleaner this way, thanks Daniel.

File: transformers/modeling_utils.py (hunk 1 above)
Comment (jysohn23) [reply transformers/modeling_utils.py:257]: As far as I know, they don't have such tagger but let me see if they do. Good idea we could even upstream if they don't. Thanks.

File: examples/run_glue_tpu.py (hunk 2 above)
Comment (jysohn23) [reply examples/run_glue_tpu.py:359]: Yep, correct.

**Pull request merged**